tools: Improve install-monorepo.sh script

anomiex opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
The big change here is adding some better error handling around the attempt to `jetpack cli link`. If `pnpm setup` hasn't been done successfully, `pnpm link --global` will silently fail and so `jetpack cli link` will too.

Beyond that, this also pulls in our "chalk-lite" library and uses those functions to format the output a little nicer.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->
p1691445603197599-slack-CDLH4C1UZ

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
No

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Ideally, try this on a fresh Mac that doesn't have pnpm or anything installed yet.

anomiex committed:
tools: Improve install-monorepo.sh script

The big change here is adding some better error handling around the
attempt to `jetpack cli link`. If `pnpm setup` hasn't been done
successfully, `pnpm link --global` will silently fail and so
`jetpack cli link` will too.

Beyond that, this also pulls in our "chalk-lite" library and uses those
functions to format the output a little nicer.

kraftbj commented:
In testing it on a fresh Mac, it installed `pnpm`, but `pnpm` wasn't yet available. There's a notice to run `source ~/.zshrc` to enable pnpm. A fresh terminal window does work, so that makes sense.

kraftbj commented on tools/install-monorepo.sh:
```diff
@@ -18,102 +19,100 @@ if [[ $1 ]]; then
-	echo "Installing pnpm"
-	curl -f https://get.pnpm.io/v6.16.js | node - add --global pnpm
+	info "Installing pnpm"
+	curl -fsSL https://get.pnpm.io/install.sh | sh -
```
```suggestion
	curl -fsSL https://get.pnpm.io/install.sh | sh -
	source ~/.zshrc
```

Adding this here resolved the issue for me.

anomiex replied:
This will not work for people using bash instead of zsh. Or dash or fish or ksh.

For that matter it might not always work for people running zsh, if their `~/.zshrc` includes stuff that's not compatible with the bash that the current script is running with.

... So many different ways to install JS crap, and they all suck.

gravityrail replied:
Would this work?
```
if [[ $0 == "zsh" ]]; then
  source ~/.zshrc
fi
...
```

gravityrail replied:
It wouldn't solve all your problems, but personally this PR is an improvement for me and I'd like to get it merged :)

anomiex replied:
No. But c766a4f might.

kraftbj reviewed (changes requested):
One tweak to have the script sing.

anomiex committed:
Switch back to old method of installing pnpm

So many ways to do it, and they all suck.

* This way is using an old version of pnpm. Someday that'll probably
  break.
* https://get.pnpm.io/install.sh relies on adding a snippet to the login
  shell path to work at all.
* corepack will lose pnpm any time nvm installs a new version.
* Who knows which version brew might have?

Also, after running `pnpm setup`, try reading the PNPM_HOME out of the
login shell so `jetpack cli link` has a chance of working later.

anomiex committed:
Merge remote-tracking branch 'origin/trunk' into update/install-monorepo-script

anomiex committed:
Merge remote-tracking branch 'origin/trunk' into update/install-monorepo-script
